projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DockerRegistryService.java

```diff
@@ -34,6 +59,68 @@ public String builderImage() {
 
   // TODO: Implement
   public boolean dataJobImageExists(String imageName) {
+
+    if (registryType.equalsIgnoreCase("ecr")) {
+      return checkImageExistsInEcr(imageName);
+    }
+
+    return false;
+  }
+
+  private boolean checkImageExistsInEcr(String imageName) {
+    // imageName is a string of the sort:
+    // 850879199482.dkr.ecr.us-west-2.amazonaws.com/sc/dp/integration-test-6ebd50b6:b722432f9f0aaa43fcec6cb68993651145815990
+    String imageRepoTag = imageName.split("amazonaws.com/")[1];
+    ImageIdentifier imageIdentifier =
+        new ImageIdentifier().withImageTag(imageRepoTag.split(":")[1]);
+    String imageRepository = imageRepoTag.split(":")[0];
+
+    var credentials = awsCredentialsService.createTemporaryCredentials();
+
+    AmazonECR ecrClient;
+    AWSStaticCredentialsProvider awsStaticCredentialsProvider;
+    if (!credentials.awsSessionToken().isBlank()) {
+      // need to include session token
+      var sessCreds =
+          new BasicSessionCredentials(
+              credentials.awsAccessKeyId(),
+              credentials.awsSecretAccessKey(),
+              credentials.awsSessionToken());
+      awsStaticCredentialsProvider = new AWSStaticCredentialsProvider(sessCreds);
+
+    } else {
+      // otherwise, we auth without session token
+      var creds =
+          new BasicAWSCredentials(credentials.awsAccessKeyId(), credentials.awsSecretAccessKey());
+      awsStaticCredentialsProvider = new AWSStaticCredentialsProvider(creds);
+    }
+
+    ecrClient =
+        AmazonECRClientBuilder.standard()
+            .withCredentials(awsStaticCredentialsProvider)
+            .withRegion(credentials.region())
+            .build();
+
+    DescribeImagesRequest describeImagesRequest =
+        new DescribeImagesRequest()
+            .withRepositoryName(imageRepository)
+            .withImageIds(imageIdentifier);
+
+    try {
+      DescribeImagesResult describeImagesResult = ecrClient.describeImages(describeImagesRequest);
+      if (describeImagesResult.getImageDetails().size() == 1) {
+        return true;
+      }
+    } catch (ImageNotFoundException
+        | RepositoryNotFoundException
+        | ServerException
+        | InvalidParameterException e) {
+      log.info("Could not find image due to: {}", e);
+      return false;
+    } catch (Exception e) {
+      log.error("Failed to check if image exists due to: ", e);
+      return false;
+    }
```

Comment:
Catching the above exception (ImageNotFoundException, ,...) make sense since they indicate that image does not exists. 

This exception though is some type of error and it seems more appropriate the caller of this method to handle it since it knows best if it should re-try, ignore it, abort.